app/scripts/modules/core/src/application/nav/NavCategory.tsx

```diff
@@ -1,19 +1,20 @@
 import React from 'react';
 
 import { DataSourceNotifications } from 'core/entityTag/notifications/DataSourceNotifications';
-import { Icon, useDataSource } from '../../presentation';
+import { Icon, Tooltip, useDataSource } from '../../presentation';
 
 import { ApplicationDataSource } from '../service/applicationDataSource';
 import { Application } from '../application.model';
 import { IEntityTags } from '../../domain';
 
 export interface INavCategoryProps {
+  app: Application;
   category: ApplicationDataSource;
   isActive: boolean;
-  app: Application;
+  isCollapsed: boolean;
 }
 
-export const NavCategory = ({ app, category, isActive }: INavCategoryProps) => {
+export const NavCategory = ({ app, category, isActive, isCollapsed }: INavCategoryProps) => {
   const { alerts, badge, iconName, key, label } = category;
 
   const { data: badgeData } = useDataSource(app.getDataSource(badge || key));
```

Comment:
this is really confusing to me.  If category is already a DataSource why are we getting the DataSource here?

Reply:
For some data sources, the "executable" versions of that data source are gathered differently. For example, a pipeline's data comes from the `executions` dataSources, but the number of running pipelines comes from `runningExecutions`. The `runningExecutions`, are a unique data source in itself.

This confused me at first too, but this is how the dataSource is configured initially. There is an attribute called `badge`, which stores the key of the supplementary data source that has the executable tasks.